sympy/matrices/__init__.py

```diff
@@ -60,4 +90,28 @@
     'det', 'diagonalize_vector', 'DiagMatrix', 'DiagonalMatrix',
     'DiagonalOf', 'trace', 'DotProduct', 'kronecker_product',
     'KroneckerProduct', 'PermutationMatrix', 'MatrixPermute',
+
+    '_find_reasonable_pivot', '_find_reasonable_pivot_naive',
+    '_berkowitz_toeplitz_matrix',
+    '_adjugate', '_charpoly', '_cofactor', '_cofactor_matrix', '_det',
+    '_det_bareiss', '_det_berkowitz', '_det_LU', '_minor', '_minor_submatrix',
+    'adjugate', 'charpoly', 'cofactor', 'cofactor_matrix', 'det', 'det_bareiss',
+    'det_berkowitz', 'det_LU', 'minor', 'minor_submatrix',
```

Comment:
There needs to be careful consideration of which functions are actually intended to make it up to top-level. I see that these are only going to `sympy.matrices` and not `sympy` but I still don't think all of these belong here. Any name that begins with an underscore is implicitly marked as internal so either the name should not have an underscore or it should not be here (many of these should not be here I think).

Reply:
Changed...